File: Domain/Aggregates/Post.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;
using Domain.Entities;
using Domain.Models;

namespace Domain.Aggregates
{
    [Table("Posts")]
    public class Post : AggregateRoot<int>
    {
        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public virtual User? User { get; set; }
        public PostType Type { get; set; }
        public string? Description { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public int? WorkoutId { get; set; }
        [ForeignKey("WorkoutId")]
        public virtual Workout? Workout { get; set; }
        public int? WorkoutPlanId { get; set; }
        [ForeignKey("WorkoutPlanId")]
        public virtual WorkoutPlan? WorkoutPlan { get; set; }
        public virtual ICollection<PostLike> Likes { get; set; } = [];
        public virtual ICollection<Comment> Comments { get; set; } = [];
    }
}

File: UI/src/app/core/models/Post.ts
import { Comment } from './Comment';
import { PostType } from './Enums';
import { Workout } from './Workout';

export interface Post {
  id: number;
  userId: number;
  type: PostType;
  description: string | null;
  createdDate: string;
  updatedDate: string | null;
  workoutId: number | null;
  workoutPlanId: number | null;
  workout: Workout | null;
  likesCount: number;
  comments: Comment[];
}

export interface PostCreate {
  type: PostType;
  description: string | null;
  workoutId: number | null;
  workoutPlan: number | null;
}

export interface PostUpdate {
  description: string | null;
}
